File: kernel/src/test/java/com/sos/scheduler/engine/kernel/test/ResourcesAsFilesProviderTest.java
package com.sos.scheduler.engine.kernel.test;

import static com.sos.scheduler.engine.kernel.util.Classes.springPattern;
import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.containsInAnyOrder;

import java.io.File;

import org.junit.Rule;
import org.junit.Test;
import org.junit.rules.TemporaryFolder;

import com.google.common.collect.ImmutableMap;

public class ResourcesAsFilesProviderTest {
    private static final String[] expectedFilenames = {"scheduler.xml", "factory.ini", "sos.ini"};

    @Rule public final TemporaryFolder folder = new TemporaryFolder();

    @Test public void test() {
        File dir = folder.getRoot();
        ImmutableMap<String,ResourceFile> map = ResourcesAsFilesProvider.provideResourcesAsFiles(springPattern(ResourcesAsFilesProviderTest.class.getPackage(), "config/*"), dir);
        assertThat(map.keySet(), containsInAnyOrder(expectedFilenames));
//        String[] list = dir.list();
//        assertThat(list, arrayContainingInAnyOrder(expectedFilenames));
    }
}

File: engine-test/src/test/scala/com/sos/scheduler/engine/test/binary/ResourcesAsFilesProviderTest.scala
package com.sos.scheduler.engine.test.binary

import com.google.common.io.Files.createTempDir
import com.sos.scheduler.engine.common.system.Files.removeDirectoryRecursivly
import com.sos.scheduler.engine.test.binary.ResourcesAsFilesProvider.provideResourcesAsFiles
import org.junit.runner.RunWith
import org.scalatest.FreeSpec
import org.scalatest.Matchers._
import org.scalatest.junit.JUnitRunner
import org.springframework.core.io.support.PathMatchingResourcePatternResolver

@RunWith(classOf[JUnitRunner])
final class ResourcesAsFilesProviderTest extends FreeSpec {

  "provideResourcesAsFiles" in {
    val directory = createTempDir()
    try {
      val resources = (new PathMatchingResourcePatternResolver).getResources("classpath*:com/sos/scheduler/engine/test/files/config/*")
      provideResourcesAsFiles(resources, directory).keySet shouldEqual Set("scheduler.xml", "factory.ini", "sos.ini")
    }
    finally removeDirectoryRecursivly(directory)
  }
}

